src/main/scala/com/cognite/sdk/scala/v1/propertyMap.scala

```diff
@@ -47,15 +47,40 @@ object PropertyMap {
       case _ => true
     }
 
+  // scalastyle:off cyclomatic.complexity method.length
```

Comment:
Move `toPropertyString` to its own `private def`, and you shouldn't need to use those.
I think the whole `dmp.`type` match {` thing could also benefit from being moved to a helper method.

Reply:
done.